### PR TITLE
Stream 9: Add end-to-end streaming example.

### PR DESCRIPTION
**Pull request by willhug**
Summary: This PR adds an example of how to use grpc streaming, it's not
entirely necessary but was very useful for quickly testing the
grpc/proto stream apis before a lot of the test infra was finished.

### CODE REVIEW

**Comment by peter-edge** [comment]
@willhug I added a `service-yest.yaml` setup so this is used for testing.

File: internal/examples/streaming/client/main.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	stream, err := client.HelloThere(ctx, yarpc.WithHeader("test", "testtest"))
+	if err != nil {
+		return fmt.Errorf("failed to create stream: %s", err.Error())
+	}
+
+	scanner := bufio.NewScanner(os.Stdin)
+	fmt.Printf(">>> ")
```
Comment (mway) [inline internal/examples/streaming/client/main.go]: consider `defer fmt.Println("")` or similar to flush the final newline

File: internal/examples/streaming/client/main.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		log.Fatal(err)
+	}
+}
+
+func do() error {
+	dispatcher := yarpc.NewDispatcher(yarpc.Config{
+		Name: "keyvalue-client",
+		Outbounds: yarpc.Outbounds{
+			"keyvalue": {
+				Stream: grpc.NewTransport().NewSingleOutbound("127.0.0.1:24039"),
```
Comment (mway) [inline internal/examples/streaming/client/main.go]: use `:0` instead

File: internal/examples/streaming/server/main.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+func main() {
+	if err := do(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func do() error {
+	var inbound transport.Inbound
+	listener, err := net.Listen("tcp", "127.0.0.1:24039")
```
Comment (mway) [inline internal/examples/streaming/server/main.go]: use `:0` instead

**Pull request merged**